### PR TITLE
Serialize join_token to group admins

### PR DESCRIPTION
**Pull request by marten**

### CODE REVIEW

File: app/models/user_group.rb [hunk 1]
```diff
@@ -79,6 +79,14 @@ def owns?(resource)
                             resource_type: resource.class.to_s)
   end
 
+  def has_admin?(user)
+    membership_for_user(user).try { |m| m.roles.include?("group_admin") }
+  end
+
+  def membership_for_user(user)
+    memberships.find_by(user_id: user.id)
```
Comment (camallen) [inline app/models/user_group.rb]: i assume this will find the right one from memberships?

File: app/models/user_group.rb [hunk 2]
```diff
@@ -79,6 +79,14 @@ def owns?(resource)
                             resource_type: resource.class.to_s)
   end
 
+  def has_admin?(user)
+    membership_for_user(user).try { |m| m.roles.include?("group_admin") }
```
Comment (camallen) [inline app/models/user_group.rb]: should this method belong in the `memberships` model?

File: app/models/user_group.rb [hunk 3]
```diff
@@ -79,6 +79,14 @@ def owns?(resource)
...
+  end
+
+  def membership_for_user(user)
+    memberships.find_by(user_id: user.id)
+  end
+
   def identity?
     !!memberships.where(identity: true).pluck(:identity).first
```
Comment (camallen) [inline app/models/user_group.rb]: Just looking at this and thinking would exists work here?

**Comment by camallen** [comment]
Looks good, just a couple of comments.

File: app/models/user_group.rb (hunk 1 above)
Comment (marten) [inline app/models/user_group.rb]: Membership has a unique index on (user_id, user_group_id) so can only ever be one membership.

**Pull request merged**